R/utils.R

```diff
@@ -151,3 +151,12 @@ anyUnnamed <- function(x) {
   # List with name attribute; check for any ""
   any(!nzchar(nms))
 }
+
+# Get the source filename out of a script or similar
+# entry.
+pluck_src <- function(x) {
+  if (is.character(x)) return(x)
+  if (is.list(x) && "src" %in% names(x)) return(x$src)
+  if (is.list(x)) return(vapply(x, pluck_src, ""))
```

Comment:
I'm finding it a bit challenging to understand what exactly this function is trying to do. I think it's in part because the shape of the data going in is pretty loose -- that's something that has long been a cause of issues when dealing with html dependency objects.

In this particular case, I'm wondering about the `NULL` return and the `vapply()`. If this `vapply()` is called on the object and one of the items returns `NULL`, that will result in an error. Will it ever get to the `NULL` at the end of the function? If not, it would make sense to have it throw an error with a helpful message.